pkg/landscape/app/metadata-push-hook.hoon

```diff
@@ -82,7 +101,44 @@
     ==
   --
 ::
-++  on-agent  on-agent:def
+++  on-agent
+  |=  [=wire =sign:agent:gall]
+  ?.  ?=([%groups ~] wire)
+    (on-agent:def wire sign)
+  ?+  -.sign  (on-agent:def wire sign)
+    %kick  :_(this ~[watch-groups:hc])
+  ::
+      %fact
+    ?.  =(p.cage.sign %group-update-0)  `this
+    =+  !<(=update:group-store q.cage.sign)
+    ?.  ?=(%remove-members -.update)  `this
+    |^
+    =/  graphs=(set resource)
+      (hosting-graphs resource.update)
+    :_  this
+    %+  weld
+       (turn ~(tap in graphs) (cury revoke %graph-push-hook))
+    ?.  =(entity.resource.update our.bowl)  ~
+    (revoke %metadata-push-hook resource.update)^~
+    ::
+    ++  revoke
+      |=  [=dude:gall rid=resource]
+      =/  =action:push-hook  [%revoke ships.update rid]
+      =/  =cage              push-hook-action+!>(action)
+      (poke-our:pass dude cage)
+    ::
+    ++  hosting-graphs
+      |=  rid=resource
+      ^-  (set resource)
+      =/  graphs=associations:store
+        (app-metadata-for-group:met resource.update %graph)
+      %-  ~(gas in *(set resource))
+      %+  murn  ~(tap in ~(key by graphs))
+      |=  [app=term graph=resource]
+      ?.  =(our.bowl entity.graph)  ~
+      `graph
```

Comment:
Using `+skim` (or `+skip`) might be semantically cleaner.

Reply:
using the filtering functions would necessitate another transform, which imo kinda negates any improvement in clarity